### PR TITLE
feat: add default maximum transaction fee management to Client

### LINKED ISSUE #542: [Beginner]: Add `defaultMaxTransactionFee` getter and setter to Client
## 🐥 Beginner Friendly

This issue is a great fit for contributors who are ready to explore the Hiero Swift codebase a little more and take on slightly more independent work.

Beginner Issues often involve reading existing Swift code, understanding how different parts of the SDK fit together, and making small, thoughtful updates that follow established patterns.

The goal is to support skill growth while keeping the experience approachable, well-scoped, and enjoyable.

## 👾 Description of the Issue

The `Client` class in `Sources/Hiero/Client/Client.swift` has an internal `_maxTransactionFee` property that stores the maximum transaction fee in tinybars, and an internal `maxTransactionFee` computed property that converts it to `Hbar`. However, there's no public way for users to get or set this default value.

Other Hiero SDKs (Java, Go, JavaScript, Rust) provide public `getDefaultMaxTransactionFee()` and `setDefaultMaxTransactionFee()` methods. This allows users to configure a default maximum fee that applies to all transactions unless explicitly overridden per-transaction.

## 💡 Proposed Solution

Add public getter and setter methods for the default maximum transaction fee:

1. `getDefaultMaxTransactionFee() -> Hbar?` - Returns the current default, or `nil` if not set
2. `setDefaultMaxTransactionFee(_ fee: Hbar)` - Sets the default maximum fee

The implementation should:
- Use the existing `_maxTransactionFee` atomic storage
- Convert between `Int64` (tinybars) and `Hbar` appropriately
- Validate that the fee is non-negative
- Follow the existing pattern used by other getters/setters in the Client class

## 👩‍💻 Implementation Steps

To implement this feature:

- [ ] Open `Sources/Hiero/Client/Client.swift`
- [ ] Find the existing internal `maxTransactionFee` property (around line 140)
- [ ] Add public getter and setter methods:

```swift
/// Returns the default maximum transaction fee, or `nil` if unlimited.
///
/// When set, this fee is used as the default for all transactions unless
/// explicitly overridden on the transaction itself.
public func getDefaultMaxTransactionFee() -> Hbar? {
    let value = _maxTransactionFee.load(ordering: .relaxed)
    guard value != 0 else { return nil }
    return .fromTinybars(value)
}

/// Sets the default maximum transaction fee for all transactions.
///
/// - Parameter fee: The maximum fee in Hbar. Must be non-negative.
/// - Returns: Self for method chaining.
/// - Throws: `HError` if the fee is negative.
@discardableResult
public func setDefaultMaxTransactionFee(_ fee: Hbar) throws -> Self {
    guard fee.toTinybars() >= 0 else {
        throw HError(kind: .invalidArgument, description: "defaultMaxTransactionFee must be non-negative")
    }
    _maxTransactionFee.store(fee.toTinybars(), ordering: .relaxed)
    return self
}
```

- [ ] Review how other SDKs handle validation (Java throws for negative values)
- [ ] Add unit tests to verify:
  - Getting returns `nil` when not set
  - Setting and getting works correctly
  - Negative values are rejected
- [ ] Build the SDK with `swift build` and ensure all tests pass

**Reference implementations:**
- Java: `Client.getDefaultMaxTransactionFee()` / `setDefaultMaxTransactionFee(Hbar)`
- Go: `Client.GetDefaultMaxTransactionFee()` / `SetDefaultMaxTransactionFee(Hbar)`
- Rust: `Client.default_max_transaction_fee()` / `set_default_max_transaction_fee(Hbar)`

## ✅ Acceptance Criteria

To help get this change merged smoothly:

- [ ] **Scope:** Changes are limited to this issue
- [ ] **Behavior:** No other SDK behavior or API changes
- [ ] **Tests:** Existing and any new tests pass
- [ ] **Review:** All code review feedback addressed

## 📋 Step-by-Step Contribution Guide

To help keep contributions consistent and easy to review, we recommend following these steps:

- [ ] Comment `/assign` to request the issue
- [ ] Wait for assignment
- [ ] Fork the repository and create a branch
- [ ] Set up the project using the instructions in `README.md`
- [ ] Make the requested changes
- [ ] Sign each commit using `-s -S`
- [ ] Push your branch and open a pull request

Read [Workflow Guide](docs/training/workflow.md) for step-by-step workflow guidance.  
Read [README.md](README.md) for setup instructions.

❗ Pull requests **cannot be merged** without `S` and `s` signed commits.  
See the [Signing Guide](docs/training/signing.md).

## 🤔 Additional Information

If you have questions while working on this issue, feel free to ask!

You can reach the community and maintainers here:
[Hiero Discord](https://discord.com/channels/905194001349627914/1337424839761465364)

Whether you need help finding the right file, understanding existing code, or confirming your approach — we're happy to help.

### PR DESCRIPTION
**Pull request by danielmarv**
**Description**:
This pull request adds support for configuring a default maximum transaction fee in the `Client` class, along with comprehensive unit tests to verify its behavior. The main focus is on providing a fluent interface for setting and retrieving this default fee, as well as ensuring correct handling of edge cases such as negative or zero values.

Enhancements to transaction fee configuration:

* Added `getDefaultMaxTransactionFee()` and `setDefaultMaxTransactionFee(_:)` methods to the `Client` class, allowing users to get or set a default maximum transaction fee for all transactions, with validation to prevent negative values. (`Sources/Hiero/Client/Client.swift`, [Sources/Hiero/Client/Client.swiftR182-R207](diffhunk://#diff-b8218ee329547bda3d2518a5d9a007b4e8bbb4549b2bdd5c224b9c578571ba6aR182-R207))

Testing and validation:

* Introduced a suite of unit tests to verify the default behavior, correct storage, fluent interface, handling of zero values, and error throwing for negative fees in the `Client` class. (`Tests/HieroUnitTests/ClientUnitTests.swift`, [Tests/HieroUnitTests/ClientUnitTests.swiftR188-R233](diffhunk://#diff-cd59de9ffbdecc2f8f3ef79446b416f01db6d7d23d2997eec807457188c965afR188-R233))

**Related issue(s)**:

Fixes #542 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: Sources/Hiero/Client/Client.swift [hunk 1]
```diff
@@ -179,6 +179,32 @@ public final class Client: Sendable {
...
+    ///
+    /// When set, this fee is used as the default for all transactions unless
+    /// explicitly overridden on the transaction itself.
+    public func getDefaultMaxTransactionFee() -> Hbar? {
+        maxTransactionFee
+    }
+
+    /// Sets the default maximum transaction fee for all transactions.
+    ///
+    /// - Parameter fee: The maximum fee in Hbar. Must be non-negative.
```
Comment (rwalworth) [inline Sources/Hiero/Client/Client.swift:192]: **issue (blocking):** The `- Parameter fee:` line doesn't document a non-obvious behavior: setting `fee` to zero is treated as "no limit" - `getDefaultMaxTransactionFee()` will return `nil` after calling `setDefaultMaxTransactionFee(.fromTinybars(0))`.

Your test `test_SetDefaultMaxTransactionFeeZeroReturnsNilWhenRead` already covers this correctly, but the doc comment should also explain it so callers understand the contract without reading tests. Please update the doc comment to something like:

```swift
/// - Parameter fee: The maximum fee in Hbar. Must be non-negative.
///   A value of zero clears the default fee limit, equivalent to not having set one
///   (i.e., `getDefaultMaxTransactionFee()` will return `nil`).
```

**Review by rwalworth — CHANGES_REQUESTED** [review]
Thanks for tackling this, @danielmarv! The implementation is on the right track - I left one comment below that need to be addressed before this can be merged. Once it's fixed, we'll be good to go!

**Review by rwalworth — APPROVED** [review]
LGTM - you nailed the doc comment update, and the implementation looks great! 

Running the workflows now and will merge once they pass.

Congratulations on your first contribution to the Hiero Swift SDK, @danielmarv! 🎉 Thank you for taking the time to work through this!
